### PR TITLE
Extract storing length to its own function.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/codegen/ABIFunctions.h [hunk 1]
```diff
@@ -231,6 +231,11 @@ class ABIFunctions
 	/// Only works for memory arrays and storage arrays that store one item per slot.
 	std::string nextArrayElementFunction(ArrayType const& _type);
 
+	/// @returns the name of a function using in encoding that stores the length
```
Comment (axic) [inline libsolidity/codegen/ABIFunctions.h]: `function using in encoding`? 

Is this meant to be used in encoding?

**Review by axic — DISMISSED** [review]
Looks good apart from the description. Fingers crossed the inliner will remove all these no-op functions :wink:

File: libsolidity/codegen/ABIFunctions.cpp [hunk 2]
```diff
@@ -621,12 +621,12 @@ string ABIFunctions::abiEncodingFunctionCalldataArray(
...
 			// <readableTypeNameFrom> -> <readableTypeNameTo>
 			function <functionName>(start, length, pos) -> end {
-				<storeLength> // might update pos
+				pos := <storeLength>(pos, length)
 				<copyFun>(start, pos, length)
 				end := add(pos, <roundUpFun>(length))
 			}
 		)");
-		templ("storeLength", _to.isDynamicallySized() ? "mstore(pos, length) pos := add(pos, 0x20)" : "");
+		templ("storeLength", arrayStoreLengthForEncodingFunction(_to));
```
Comment (axic) [inline libsolidity/codegen/ABIFunctions.cpp]: Need to use `_toArrayType` here.

File: libsolidity/codegen/ABIFunctions.h (hunk 1 above)
Comment (chriseth) [reply libsolidity/codegen/ABIFunctions.h]: ah right, should be "used in encoding" ("for" encoding? "during" encoding?)

**Review by axic — APPROVED** [review]

**Pull request merged**